src/ophyd_async/epics/areadetector/controllers/kinetix_controller.py

```diff
@@ -0,0 +1,55 @@
+import asyncio
+from typing import Optional, Set
+
+from ophyd_async.core import AsyncStatus, DetectorControl, DetectorTrigger
+from ophyd_async.epics.areadetector.drivers.ad_base import (
+    DEFAULT_GOOD_STATES,
+    DetectorState,
+    start_acquiring_driver_and_ensure_status,
+)
+
+from ..drivers.kinetix_driver import KinetixDriver, KinetixTriggerMode
+from ..utils import ImageMode
+
+KINETIX_TRIGGER_MODE_MAP = {
+    DetectorTrigger.internal: KinetixTriggerMode.internal,
+    DetectorTrigger.constant_gate: KinetixTriggerMode.gate,
+    DetectorTrigger.variable_gate: KinetixTriggerMode.gate,
+    DetectorTrigger.edge_trigger: KinetixTriggerMode.edge,
+}
+
+
+class KinetixController(DetectorControl):
+    def __init__(
+        self,
+        driver: KinetixDriver,
+        good_states: Set[DetectorState] = set(DEFAULT_GOOD_STATES),
+    ) -> None:
+        self.driver = driver
+        self.good_states = good_states
+
+    def get_deadtime(self, exposure: float) -> float:
+        return 0.001
+
+    async def arm(
+        self,
+        num: int,
+        trigger: DetectorTrigger = DetectorTrigger.internal,
+        exposure: Optional[float] = None,
+    ) -> AsyncStatus:
+        await asyncio.gather(
+            self.driver.trigger_mode.set(KINETIX_TRIGGER_MODE_MAP[trigger]),
+            self.driver.num_images.set(999_999 if num == 0 else num),
```

Comment:
The `999999` number is the max images that Pilatus can make, is there a similar limit for the kinetix?

Reply:
I think the limit for the Kinetix is just the max value of an asynInt32 record, but I felt like 999999 is probably large enough so I kept it in. I have no issue with removing this (although the Kinetix is fast enough with a large enough frame size that the filesaving plugin is almost certainly not going to keep up, so you will run out of memory before you reach 999999 images).